src/main/java/org/mule/module/json/internal/JsonSchemaParser.java

```
@@ -65,4 +89,69 @@ public static JsonNode getSchemaJsonNode(String schemaContent, String schemaLoca
       throw new ModuleException(format("Malformed Json Schema: %s", e.getMessage()), INVALID_INPUT_JSON);
     }
   }
+
+  private static boolean isSelfReferencingOrInternal(String schemaUrl) {
+    try {
+      URL url = new URL(schemaUrl);
+      String host = url.getHost().toLowerCase(Locale.ROOT);
+      Boolean cached = SELF_REF_CACHE.get(host);
+      if (cached != null) {
+        return cached;
+      }
+
+      if ("localhost".equals(host) || KNOWN_SELF_HOSTS.contains(host)
+          || (LOCAL_IP_PATTERN.matcher(host).matches() || LOCAL_IPV6_PATTERN.matcher(host).matches())) {
+        SELF_REF_CACHE.put(host, true);
+        return true;
+      }
+
+
+      // DNS resolution for more complex cases
+      InetAddress address = InetAddress.getByName(host);
+      boolean result = address.isAnyLocalAddress() // covers 0.0.0.0
+          || address.isLoopbackAddress() // 127.x.x.x
+          || address.isSiteLocalAddress() // 192.168.x.x, 10.x.x.x, etc.
+          || KNOWN_SELF_HOSTS.contains(address.getHostAddress());
+
+      SELF_REF_CACHE.put(host, result);
+      return result;
+
+    } catch (MalformedURLException | UnknownHostException e) {
+      // Cache negative result for this host
+      String host = null;
+      try {
+        host = new URL(schemaUrl).getHost().toLowerCase(Locale.ROOT);
+      } catch (Exception ignore) {
+        //ignore
```

We shouldn't have empty catch blocks, at least leave an error log, for future debugging.